### ISSUE TITLE
az_json_writer_append_json_text doesn't add comma when needed

### ISSUE THREAD

**Issue by kevinior** [Azure/azure-sdk-for-c#1574]
If I'm writing a JSON array containing nested JSON then no commas are inserted between the array elements:
```c
TEST_F(Azure, NestedJson) {
    char Buffer[1024];
    char Result[1024];
    az_json_writer Writer = { 0 };
    EXPECT_EQ(AZ_OK, az_json_writer_init(&Writer, AZ_SPAN_FROM_BUFFER(Buffer), NULL));
    EXPECT_EQ(AZ_OK, az_json_writer_append_begin_array(&Writer));
    EXPECT_EQ(AZ_OK, az_json_writer_append_json_text(&Writer, AZ_SPAN_FROM_STR("{\"a\": 1}")));
    EXPECT_EQ(AZ_OK, az_json_writer_append_json_text(&Writer, AZ_SPAN_FROM_STR("{\"b\": 2}")));
    EXPECT_EQ(AZ_OK, az_json_writer_append_end_array(&Writer));
    az_span_to_str(Result, sizeof(Result), az_json_writer_get_bytes_used_in_destination(&Writer));
    EXPECT_STREQ("[{\"a\": 1},{\"b\": 2}]", Result);
}
```

The result I'd expect is `[{"a": 1},{"b": 2}]` but instead I get `[{"a": 1}{"b": 2}]`.

I'm using version 1.1.0-beta.3 of the SDK.

**Comment by RickWinter** [comment]
FYI: @ahsonkhan

**Comment by ahsonkhan** [comment]
Great catch, @kevinior, fixed in https://github.com/Azure/azure-sdk-for-c/pull/1600

**Comment by ahsonkhan** [comment]
@RickWinter I think its worth considering porting this fix as a hot-fix for the 1.0.0 release, and make a 1.0.1 patch version bump.  This API (and bug) was introduced prior to 1.0.0 GA (specifically in preview 5 - https://github.com/Azure/azure-sdk-for-c/releases/tag/1.0.0-preview.5)

https://github.com/Azure/azure-sdk/blob/0cd54f07c99a9534036f8a1a376bba7064f0b1b1/docs/policies/repobranching.md#hotfix-branches
